packet/http.go

```diff
@@ -56,12 +57,16 @@ func ParseUrl(raw []byte) {
 
 }
 
-func NewHttpPacket(raw []byte) (*HttpPacket, error) {
-	pkt := &HttpPacket{raw: raw}
-
-	pkt.parse()
-
-	return pkt, nil
+func NewHttpPacketFromReader(rdr io.Reader) (*HttpPacket, error) {
```

Comment:
Previously, the request was parsed from a pre-read buffer, the size of which is 0 <= n < 1024. There could have been a problem if the request was sent in two parts. A reader was created, based on the underlying connection."